app/lib/feed_manager.rb

```diff
@@ -230,6 +230,36 @@ def populate_home(account)
     end
   end
 
+  # Completely clear multiple feeds at once
+  # @param [Symbol] type
+  # @param [Array<Integer>] ids
+  # @return [void]
+  def clean_feeds!(type, ids)
+    reblogged_id_sets = {}
+
+    redis.pipelined do
+      ids.each do |feed_id|
+        redis.del(key(type, feed_id))
+        reblog_key = key(type, feed_id, 'reblogs')
+        # We collect a future for this: we don't block while getting
+        # it, but we can iterate over it later.
+        reblogged_id_sets[feed_id] = redis.zrange(reblog_key, 0, -1)
+        redis.del(reblog_key)
+      end
+    end
+
+    # Remove all of the reblog tracking keys we just removed the
+    # references to.
+    redis.pipelined do
+      reblogged_id_sets.each do |feed_id, future|
+        future.value.each do |reblogged_id|
```

Comment:
How does getting the future interact with being inside another pipelined block?

Reply:
Hm, I haven't changed anything here (it's code moved from the cleanup scheduler).

There are two pipelines, one after the other, I think the benefit is that we can build the first one directly, and then wait for its response when building the second one.